r3bgen/R3BBeamProperties.cxx

```diff
@@ -0,0 +1,40 @@
+/******************************************************************************
+ *   Copyright (C) 2019 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2019 Members of R3B Collaboration                          *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+#include "R3BBeamProperties.h"
+#include "R3BDistribution2D.h"
+#include "R3BDistribution3D.h"
+
+#include "TGeoPhysicalConstants.h" // for  amu_c2
+
+constexpr auto AMU_MeV = TGeoUnit::amu_c2 * TGeoUnit::MeV; // MeV
+
+R3BBeamProperties::R3BBeamProperties()
+    : fVertex_cm(R3BDistribution3D::Delta(0, 0, 0))
+    , fSpread_mRad(R3BDistribution2D::Delta(0, 0))
+    , fBeta(R3BDistribution1D::Delta(0.5))
+{
+}
+
+void R3BBeamProperties::SetEnergyDistribution(R3BDistribution<1> energyDistribution_AMeV, int nNucleons, double mass)
+{
+    const auto nucleonMass = (mass == 0. ? nNucleons * AMU_MeV : mass);
```

Comment:
You could also use the Particle Selector (or just the mass lookup) here, but this should be good enough.